src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -187,6 +200,8 @@ void subscribeNext() {
         private final ConcatSubscriber<T> parent;
         private final AtomicInteger once = new AtomicInteger();
```

Comment:
internally, `AtomicBoolean` is implemented the same as AtomicInteger, but sure, can be changed to `AtomicBoolean`, by a separate PR.